Tests/DefaultXpringClientTest.swift

```diff
@@ -5,11 +5,43 @@ final class DefaultXpringClientTest: XCTestCase {
 
   // MARK: - Balance
 
-  func testGetBalance() {
-    // GIVEN a XpringClient.
-    let xpringClient = DefaultXpringClient()
+  func testGetBalanceWithSuccess() {
+    // GIVEN a Xpring client which will successfully return a balance from a mocked network call.
+    let xpringClient = DefaultXpringClient(networkClient: FakeNetworkClient.successfulFakeNetworkClient)
 
-    // WHEN the balance is retrieved THEN an error is thrown.
+    // WHEN the balance is requested.
+    guard let balance = try? xpringClient.getBalance(for: .destinationAddress) else {
```

Comment:
This is using shorthand. Because the `getBalance` method takes a `String`  parameter, we can infer that `.destinationAddress` is really a static property on `String`. 

We could write this line as `xpringClient.getBalance(for: String.destinationAddress)` in its long form. 

Two things that you may be curious about:
1) If you inspect the definition of `getBalance` you'll see it actually [takes an Address](https://github.com/xpring-eng/XpringKit/pull/89/files/f78e78737827e2ca4d70aa5abac94520206892d0#diff-e7065b7111b94aba4e429fa86d064b94R28) as a parameter. `Address` is really just a type alias for `String` (https://github.com/xpring-eng/XpringKit/blob/master/XpringKit/Address.swift) which is to say that they are the same object and we've just renamed it to make it more clear what it is. 

2) You may be wondering why the builtin `String` class conveniently has a properly formatted `destinationAddress` property on it. We've [actually added this ourselves](https://github.com/xpring-eng/XpringKit/blob/master/Tests/Legacy/LegacyDefaultXpringClientTest.swift#L9) using Swift extensions. Extensions let you add features to existing classes ([more on that](https://docs.swift.org/swift-book/LanguageGuide/Extensions.html)). 

A common pattern for test cases is to add static properties via extensions and then reference them as named shorthand things here. 

So to pull it together, this line could be written as:
```swift
/// Use a string literal
xpringClient.getBalance(for: "XVfC9CTCJh6GN2x8bnrw3LtdbqiVCUFyQVMzRrMGUZpokKH")

/// Use the static property defined via extension of String
xpringClient.getBalance(for: String.destinationAddress)

/// Use the same static property, but call the static object Address (recall Address is typealiased to String)
xpringClient.getBalance(for: Address.destinationAddress)

/// Let Swift infer the class the static property is on
xpringClient.getBalance(for: .destinationAddress)
```